testdata/union_enum_types.txtar

```diff
@@ -0,0 +1,39 @@
+-- cue --
+
+#StringEnum: "a" | "b" | "c" @cuetsy(kind="enum")
+#StringEnumWithMemberTypes: "a" | "b" | "c" @cuetsy(kind="enum",memberNames="First|Second|Third")
+#IntEnum: 1 | 2 | 3 @cuetsy(kind="enum",memberNames="First|Second|Third")
+
+#Expressions: {
+   sEnum: (#StringEnum & "a") | (#StringEnum & "b")
+   sEnumMem: (#StringEnumWithMemberTypes & "a") | (#StringEnumWithMemberTypes & "b")
+   iEnum: (#IntEnum & 1) | (#IntEnum & 2) | (#IntEnum & 3)
+   normal: #StringEnum & "a"
```

Comment:
let's add a test case for a nested field
```suggestion
   normal: #StringEnum & "a"
   nested: {
      nestedEnum: #StringEnum & "a"
   }
```

Reply:
Just fixed. Also I added union inside the nested function.